sound/soc/sof/mediatek/mt8195/mt8195.c

```diff
@@ -0,0 +1,337 @@
+		return ret;
+	}
+
+	adsp->pa_dram = (phys_addr_t)res.start;
+	adsp->dramsize = (u32)resource_size(&res);
+	if (((u32)adsp->pa_dram) & DRAM_REMAP_MASK) {
+		dev_err(dev, "adsp memory(0x%x) is not 4K-aligned\n",
+			(u32)adsp->pa_dram);
+		return -EINVAL;
+	}
+
+	if (adsp->dramsize < TOTAL_SIZE_SHARED_DRAM_FROM_TAIL) {
+		dev_err(dev, "adsp memory(0x%x) is not enough for share\n",
+			(u32)adsp->dramsize);
+		return -EINVAL;
+	}
+
+	dev_dbg(dev, "[ADSP] dram pbase=%pa, dramsize=0x%x\n",
+		&adsp->pa_dram, adsp->dramsize);
+
+	/* Parse CFG base */
+	mmio = platform_get_resource(pdev, IORESOURCE_MEM, 0);
+	if (!mmio) {
+		dev_err(dev, "no ADSP-CFG register resource\n");
+		return -ENXIO;
+	}
+	/* remap for DSP register accessing */
+	adsp->va_cfgreg = devm_ioremap_resource(dev, mmio);
+	if (IS_ERR(adsp->va_cfgreg))
+		return PTR_ERR(adsp->va_cfgreg);
+
+	adsp->pa_cfgreg = (phys_addr_t)mmio->start;
+	adsp->cfgregsize = (u32)resource_size(mmio);
+
+	dev_dbg(dev, "[ADSP] cfgreg-vbase=%p, cfgregsize=0x%x\n",
+		adsp->va_cfgreg, adsp->cfgregsize);
+
+	/* Parse SRAM */
+	mmio = platform_get_resource(pdev, IORESOURCE_MEM, 1);
+	if (!mmio) {
+		dev_err(dev, "no SRAM resource\n");
+		return -ENXIO;
+	}
+
+	adsp->pa_sram = (phys_addr_t)mmio->start;
+	adsp->sramsize = (u32)resource_size(mmio);
+	if (adsp->sramsize < TOTAL_SIZE_SHARED_SRAM_FROM_TAIL) {
+		dev_err(dev, "adsp SRAM(0x%x) is not enough for share\n",
+			(u32)adsp->sramsize);
+		return -EINVAL;
+	}
+
+	dev_dbg(dev, "[ADSP] sram pbase=%pa,0x%x\n", &adsp->pa_sram, adsp->sramsize);
+
+	/* Parse MBOX base */
+	for (i = 0; i < DSP_MBOX_NUM; i++) {
+		mmio = platform_get_resource(pdev, IORESOURCE_MEM, 2 + i);
+		if (!mmio) {
+			dev_err(dev, "no MBOX resource %d\n", i);
+			return -ENXIO;
+		}
+
+		adsp->pa_mboxreg[i] = (phys_addr_t)mmio->start;
+		adsp->va_mboxreg[i] = devm_ioremap_resource(dev, mmio);
+		if (IS_ERR(adsp->va_mboxreg[i]))
+			return PTR_ERR(adsp->va_mboxreg[i]);
+
+		dev_dbg(dev, "[ADSP]MBOX%d pa_mboxreg:%pa, va_mboxreg:%p\n",
+			i, &adsp->pa_mboxreg[i], adsp->va_mboxreg[i]);
+	}
+
+	return ret;
+}
+
+static int adsp_sram_power_on(struct device *dev, bool val)
+{
+	void __iomem *va_dspsysreg;
+	u32 srampool_con;
+
+	va_dspsysreg = devm_ioremap(dev, ADSP_SRAM_POOL_CON, 0x4);
+	if (!va_dspsysreg) {
+		dev_err(dev, "error: failed to ioremap sram pool base 0x%x\n",
+			ADSP_SRAM_POOL_CON);
+		return -ENOMEM;
+	}
+
+	srampool_con = readl(va_dspsysreg);
+	if (val)
+		writel(srampool_con & ~DSP_SRAM_POOL_PD_MASK, va_dspsysreg);
+	else
+		writel(srampool_con | DSP_SRAM_POOL_PD_MASK, va_dspsysreg);
+
+	return 0;
+}
+
+/*  Init the basic DSP DRAM address */
+static int adsp_memory_remap_init(struct device *dev, void *data)
+{
+	void __iomem *vaddr_emi_map;
+	struct adsp_chip_info *adsp = data;
+	int offset;
+
+	if (!adsp)
+		return -ENXIO;
+
+	vaddr_emi_map = devm_ioremap(dev, DSP_EMI_MAP_ADDR, 0x4);
+	if (!vaddr_emi_map) {
+		dev_err(dev, "error: failed to ioremap emi map base 0x%x\n",
```

Comment:
can you please remove all "error:" prefixes, this is no longer something we want, and we are gradually removing them from the existing code. For new contributions we shouldn't see those prefixes any longer. thanks!

Reply:
ok. I will remove them. thanks.